table/block_based/block_based_table_reader.cc

```diff
@@ -497,11 +502,21 @@ Status GetGlobalSequenceNumber(const TableProperties& table_properties,
           seqno_pos->second.c_str());
       return Status::Corruption(msg_buf.data());
     }
-    return Status::OK();
+    if (!ignore_seqno_in_file) {
+      return Status::OK();
+    }
+  } else {
+    version = DecodeFixed32(version_pos->second.c_str());
+    if (version > 2) {
+      return Status::Corruption(
+          ("External file version " + std::to_string(version) +
+           " is not supported. The field is likely corrupted."));
+    }
   }
 
-  uint32_t version = DecodeFixed32(version_pos->second.c_str());
-  if (version < 2) {
+  // 0 is for no version. 1 and 2 are supported versions.
+  assert(version <= 2);
```

Comment:
nit: Maybe also `assert(!ignore_seqno_in_file || version == 0)`?

Reply:
We may ingest a file from live DB, while the file itself was ingested into that DB. Then we have `ignore_seqno_in_file && version > 0`. 

I updated the above check on version number to only allow 1 and 2, and moved the updated assertion below.